[DOC] Update Tempo doc to Alloy (#3843)
grafana/tempo: docs/sources/tempo/configuration/grafana-agent/automatic-logging.md

through log messages. Well-formatted log lines are written to a Loki instance
or to `stdout` for each span, root, or process that passes through the tracing
pipeline. This allows for automatically building a mechanism for trace
discovery. On top of that, you can also get metrics from traces using Loki, and
allow quickly jumping from a log message to the trace view in Grafana.

> While this approach is useful, it isn't as powerful as [TraceQL]({{< relref
> "../../traceql" >}}). If you are here because you know you want to log the

Author: ```suggestion
While this approach is useful, it isn't as powerful as TraceQL.
If you are here because you know you want to log the
```

trace ID, to enable jumping from logs to traces, then read on!

If you want to query the system directly, read the [TraceQL

> documentation]({{< relref "../../traceql" >}}).

Reviewer: Same question here for this relref... which applies to the other relrefs as well

Author: ```suggestion
documentation](https://grafana.com/docs/tempo/<TEMPO_VERSION>/traceql).
```

> documentation](https://grafana.com/docs/tempo/<TEMPO_VERSION>/traceql).

Reviewer 2: How does `<TEMPO_VERSION>` get populated?

Author: It's set in the front matter of tempo/_index.md. If this page is backported, then any links go to the correct version of the Tempo documentation. For example, links within the 2.4 doc set stay within the 2.4 docs instead of moving to /latest..

## Before you begin

{{< admonition type="note">}}
Grafana Alloy provides tooling to convert your Agent Static or Flow configuration files into a format that can be used by Alloy.

> For more information, refer to [Migrate to Alloy]({{<relref "../grafana-alloy/migrate-alloy" >}}).

Author: ```suggestion
For more information, refer to [Migrate to Alloy]({{< relref "../grafana-alloy/migrate-alloy" >}}).
```

> For more information, refer to [Migrate to Alloy]({{< relref "../grafana-alloy/migrate-alloy" >}}).

Author: ```suggestion
For more information, refer to [Migrate to Alloy](https://grafana.com/tempo/<TEMPO_VERSION>/grafana-alloy/migrate-alloy).
```

{{< /admonition>}}

To configure automatic logging, you need to select your preferred backend and the trace data to log.

> To see all the available configuration options, refer to the [configuration reference](/docs/agent/latest/configuration/traces-config).

Author: ```suggestion
To see all the available configuration options, refer to the [configuration reference](https://grafana.com/docs/agent/latest/configuration/traces-config).
```
